### PR TITLE
Update failure checks during service provisioning

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
First part of fix for https://github.com/openshift/origin/issues/15898 to show failures in the results page of the wizard.

### CODE REVIEW

**Comment by dtaylor113** [comment]
I don't know why travis is suddenly failing on these typescript issues.   I thought if a variable didn't have a ': <whatever>'  it would default to ": any".

File: src/components/order-service/order-service.controller.ts [hunk 1]
```diff
@@ -418,7 +418,11 @@ export class OrderServiceController implements angular.IController {
       var readyCondition: any = _.find(conditions, {type: "Ready"});
 
       this.ctrl.orderComplete = readyCondition && readyCondition.status === 'True';
-      this.ctrl.error = _.find(conditions, {type: "ProvisionFailed"});
+      this.ctrl.error = _.find(conditions, function(condition: any) {
+        return condition.reason === "ProvisionFailed" ||
+               condition.reason === "ProvisionCallFailed" ||
+               condition.reason === 'ErrorCallingProvision';
+      });
```
Comment (spadgett) [inline src/components/order-service/order-service.controller.ts]: We should be checking specifically for the `Failed` condition. Something like

```js
_.some(conditions, { type: 'Failed', status: 'True' });
```

Comment (jeff-phillips-18) [inline src/components/order-service/order-service.controller.ts]: When I get the failure I am seeing:
```
type: "Ready"
status: "False"
reason: "ProvisionCallFailed"
```

Comment (spadgett) [inline src/components/order-service/order-service.controller.ts]: There should be another condition type `Failed` now. You might need to pull the latest origin-service-catalog image.

cc @pmorie to keep me honest

Comment (jeff-phillips-18) [inline src/components/order-service/order-service.controller.ts]: @pmorie Provisioning a service is failing with:

> "Provision call failed. Error provisioning Instance "whee/django-psql-persistent-4l3nw" of ServiceClass "django-psql-persistent" at Broker "template-broker": Service instance with same id but different attributes exists"

But the condition is still:
type: 'Ready'
status: 'False'
reason: 'ProvisionCallFailed'

Should the type got to 'Failed'?

Comment (jeff-phillips-18) [inline src/components/order-service/order-service.controller.ts]: I am getting the type going to 'Failed' now that I have updated to the latest origin-service-catalog.

Comment (jeff-phillips-18) [inline src/components/order-service/order-service.controller.ts]: @spadgett Fixed.

File: src/components/order-service/order-service.controller.ts [hunk 2]
```diff
@@ -418,7 +418,7 @@ export class OrderServiceController implements angular.IController {
       var readyCondition: any = _.find(conditions, {type: "Ready"});
 
       this.ctrl.orderComplete = readyCondition && readyCondition.status === 'True';
-      this.ctrl.error = _.find(conditions, {type: "ProvisionFailed"});
+      this.ctrl.error = _.find(conditions, {type: "Failed"});
```
Comment (spadgett) [inline src/components/order-service/order-service.controller.ts]: Ugh missed this before I merged, but I think it needs to be

```
this.ctrl.error = _.find(conditions, {type: "Failed", status: "True"});
```

**Pull request merged**